hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestReconWithOzoneManagerHA.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.hadoop.ozone.client.ObjectStore;
+import org.apache.hadoop.ozone.client.OzoneClientFactory;
+import org.apache.hadoop.ozone.om.OMConfigKeys;
+import org.apache.hadoop.ozone.om.OzoneManager;
+import org.apache.hadoop.ozone.recon.spi.impl.OzoneManagerServiceProviderImpl;
+import org.apache.hadoop.test.GenericTestUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+/**
+ * This class sets up a MiniOzoneHACluster to test with Recon.
+ */
+public class TestReconWithOzoneManagerHA {
+  @Rule
+  public Timeout timeout = new Timeout(300_000);
+
+  private MiniOzoneHAClusterImpl cluster;
+  private ObjectStore objectStore;
+  private final String omServiceId = "omService1";
+
+  @Before
+  public void setup() throws Exception {
+    OzoneConfiguration conf = new OzoneConfiguration();
+    conf.set(OMConfigKeys.OZONE_OM_RATIS_ENABLE_KEY, Boolean.TRUE.toString());
+    cluster = (MiniOzoneHAClusterImpl) MiniOzoneCluster.newHABuilder(conf)
+        .setClusterId(UUID.randomUUID().toString())
+        .setScmId(UUID.randomUUID().toString())
+        .setOMServiceId(omServiceId)
+        .setNumDatanodes(1)
+        .setNumOfOzoneManagers(3)
+        .includeRecon(true)
+        .setReconHttpPort(NetUtils.getFreeSocketPort())
+        .setReconDatanodePort(NetUtils.getFreeSocketPort())
+        .build();
+    cluster.waitForClusterToBeReady();
+    objectStore = OzoneClientFactory.getRpcClient(omServiceId, conf)
+        .getObjectStore();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
+
+  @Test
+  public void testReconGetsSnapshotFromLeader() throws Exception {
+    AtomicReference<OzoneManager> ozoneManager = new AtomicReference<>();
+    // Wait for OM leader election to finish
+    GenericTestUtils.waitFor(() -> {
+      OzoneManager om = cluster.getOMLeader();
+      ozoneManager.set(om);
+      return om != null;
+    }, 100, 120000);
+    Assert.assertNotNull("Timed out waiting OM leader election to finish: "
+        + "no leader or more than one leader.", ozoneManager);
+    Assert.assertTrue("Should have gotten the leader!",
+        ozoneManager.get().isLeader());
+
+    OzoneManagerServiceProviderImpl impl = (OzoneManagerServiceProviderImpl)
+        cluster.getReconServer().getOzoneManagerServiceProvider();
+
+    String hostname =
+        ozoneManager.get().getHttpServer().getHttpAddress().getHostName();
+    String expectedUrl = "http://" +
+        (hostname.equals("0.0.0.0") ? "localhost" : hostname) + ":" +
+        ozoneManager.get().getHttpServer().getHttpAddress().getPort() +
+        OZONE_OM_DB_CHECKPOINT_HTTP_ENDPOINT;
+    String snapshotUrl = impl.getOzoneManagerSnapshotUrl();
+    Assert.assertEquals("OM Snapshot should be requested from the leader.",
+        expectedUrl, snapshotUrl);
```

Comment:
Do we need to trigger an OM DB sync here and see if the sync is working?

Reply:
I intended to test only changes I made but this Test class I envision can easily extend to add more tests for HA but those can be separate Jiras, IMO.